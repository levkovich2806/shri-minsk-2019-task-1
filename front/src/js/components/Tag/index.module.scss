.tag {
  display: inline-block;
  background-color: rgba(51, 51, 51, 0.1);
  border-radius: 16px;
  padding: 0 12px;
  margin: 12px 12px 0 0;
  line-height: 22px;
  font-size: 14px;
  text-align: center;
  color: #000;
  cursor: pointer;
  max-width: 200px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;

  &:hover {
    background-color: rgba(51, 51, 51, 0.5);
  }

  &:active {
    background-color: rgba(51, 51, 51, 0.3);
  }
}
