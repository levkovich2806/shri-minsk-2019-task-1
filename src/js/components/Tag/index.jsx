import React from "react";
import styles from "./index.module.scss";

function Tag(props) {
  return <div className={styles.tag}>{props.tag}</div>;
}

export default Tag;
