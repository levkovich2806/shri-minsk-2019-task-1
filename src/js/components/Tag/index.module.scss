.tag {
  display: inline-block;
  background-color: rgba(51, 51, 51, 0.1);
  border-radius: 16px;
  padding: 0 12px;
  margin: 12px 12px 0 0;
  line-height: 22px;
  font-size: 14px;
  text-align: center;
  color: #000;
  cursor: pointer;

  &:hover {
    background-color: rgba(51, 51, 51, 0.5);
  }

  &:active {
    background-color: rgba(51, 51, 51, 0.3);
  }
}
