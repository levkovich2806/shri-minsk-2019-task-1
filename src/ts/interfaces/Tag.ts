export default interface Tag {
  id: number;
  tag: string;
}