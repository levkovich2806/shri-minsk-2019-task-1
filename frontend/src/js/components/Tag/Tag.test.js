import React from "react";
import Tag from "./index.jsx";
import renderer from "react-test-renderer";

describe("Tag", function () {
  test("Tag rendering", () => {

    const component = renderer.create(<Tag tagName="Покупки" />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
