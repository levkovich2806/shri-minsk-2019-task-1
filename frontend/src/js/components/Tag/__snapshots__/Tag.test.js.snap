// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Tag rendering 1`] = `
<div
  className="tag"
>
  Покупки
</div>
`;
